#!/usr/bin/env python
PACKAGE = "image_cloud"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("subscribe_topic_pcl",			str_t,	 	0, "topic pcl", "")
gen.add("subscribe_topic_image",		str_t,	 	0, "topic image", "")
gen.add("subscribe_topic_image_info",	str_t,	 	0, "topic image_info", "")
gen.add("publish_topic_pcl",			str_t,		0, "remapped pcl topic", "" )
gen.add("publish_topic_image",			str_t,		0, "remapped image topic", "" )
gen.add("min_color",					int_t,		0, "set minimal color value for fusion", 15, 0, 255)
gen.add("image_tf_frame_id",			str_t,	 	0, "image tf frame id", "")
gen.add("use_reference", 				bool_t, 	0, "use reference frame for time transformation and publishing", False)
gen.add("reference_frame",				str_t,	 	0, "reference frame for transformation and pointcloud publishing", "")
gen.add("queue_size",					int_t,		0, "message filter queue", 30, 1, 100 )
gen.add("tf_buffer_length",				int_t,		0, "tf buffer length in seconds", 30, 1, 500 )

exit(gen.generate(PACKAGE, "image_cloud", "fusion"))